tests/integration/test_nnswitch.py

```diff
@@ -0,0 +1,59 @@
+import os
+import pytest
+import queue
+
+from sentry_sdk.envelope import Envelope, Item, PayloadRef
+from sentry_sdk.attachments import Attachment
+
+def load_dump_file(base_file_name: str):
+    dmp_path = os.path.join(
+        os.path.dirname(__file__), "fixtures", "native", base_file_name
+    )
+
+    with open(dmp_path, "rb") as f:
+        return f.read()
+
+
+@pytest.mark.parametrize("variant", ["plain", "zstandard"])
+def test_nnswitch(
+    mini_sentry, relay_with_processing, outcomes_consumer, attachments_consumer, events_consumer, variant
+):
+    PROJECT_ID = 42
+    mini_sentry.add_full_project_config(
+        PROJECT_ID,
+        extra={"config": {}},
+    )
+    events_consumer = events_consumer()
+    outcomes_consumer = outcomes_consumer()
+    attachments_consumer = attachments_consumer()
+    relay = relay_with_processing()
```

Comment:
I think we should make that test a chain test, `relay(relay_with_processing())` to make sure stuff isn't getting double expanded etc.